fix: test curriculum challenges

ValeraS opened the pull request:
<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #XXXXX

tests only challenges for the English language

ValeraS committed:
fix: test curriculum challenges

raisedadead commented on curriculum/test-challenges.js:
```diff
@@ -189,18 +196,24 @@ function createTest({
-        throw new Error(result.error);
-      }
-    });
+Observable.fromPromise(getChallengesForLang('english'))
```
```suggestion
Observable.fromPromise(getChallengesForLang(LOCALE || 'english'))
```

We can derive the locale variable from the `.env`, let's just make it default.

ValeraS replied:
Do we need to test all languages on CI/CD?

raisedadead replied:
Yes, the languages are essentially forks of the english version, and as a matter of fact will always be behind the english versions. Meaning broken tests will have to be fixed in every language version again.

Also, the CD will be responsible for deploying the language versions of the applications to their own "hosting" locations. 

We would build these in parallel as per the language (actually its more complicated, I am over simplifying the workflow a bit).

ValeraS replied:
fixed

raisedadead reviewed (approved):
Nice work as usual @ValeraS !

raisedadead commented:
I am going to pull this down to test on my fork for the CI. Other than that we are good to go.

raisedadead commented:
@ValeraS looks like this is great.

https://github.com/raisedadead/freeCodeCamp/pull/19

raisedadead commented:
The build seems to have caught all that is required @Bouncey 	

@ValeraS 

IDK, and this not a concern of your PR per say, but can we get rid of the stack somehow:

https://travis-ci.org/raisedadead/freeCodeCamp/builds/444145777#L5144-L5156

@RandellDawson @scissorsneedfoodtoo @beaucarnes 
We would need to go thru the build and fix the linting in challenge tests. 

/cc @tchaffee

ValeraS committed:
fix(curriculum): fix broken tests

ValeraS committed:
test(curriculum): add syntax check of tests for challenges without a solution

ValeraS committed:
fix(curriculum): fix broken tests

ValeraS committed:
test(curriculum): test challenges

- test html challenges with the use of JSDOM
- check syntax of tests for modern challenges

ValeraS committed:
fix(curriculum): fix solutions of html challenges

scissorsneedfoodtoo commented:
Everything LGTM! Awesome job, @ValeraS. Just ran the test locally and everything passes without a hitch.
